python/alb_ssl_policy.py

```diff
@@ -0,0 +1,389 @@
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+
+    ###############################
+    # Add your custom logic here. #
+    ###############################
+
+	alb_arn = configuration_item['arn']
+	alb_client = get_client('elbv2',event);
+
+	listeners = alb_client.describe_listeners( LoadBalancerArns = [alb_arn,], )
+
+	for l in listeners['Listeners']:
+		if ( 'SslPolicy' in l.keys()):
+			if(l['SslPolicy'] not in COMPLIANT_POLICY):
+				return 'NON_COMPLIANT'
+
+	return 'COMPLIANT'
+
+def evaluate_parameters(rule_parameters):
+    """Evaluate the rule parameters dictionary validity. Raise a ValueError for invalid parameters.
+
+    Return:
+    anything suitable for the evaluate_compliance()
+
+    Keyword arguments:
+    rule_parameters -- the Key/Value dictionary of the Config Rules parameters
+    """
+    valid_rule_parameters = rule_parameters
+    return valid_rule_parameters
+
+####################
+# Helper Functions #
+####################
+
+# Build an error to be displayed in the logs when the parameter is invalid.
+def build_parameters_value_error_response(ex):
+    """Return an error dictionary when the evaluate_parameters() raises a ValueError.
+
+    Keyword arguments:
+    ex -- Exception text
+    """
+    return  build_error_response(internalErrorMessage="Parameter value is invalid",
+                                 internalErrorDetails="An ValueError was raised during the validation of the Parameter value",
+                                 customerErrorCode="InvalidParameterValueException",
+                                 customerErrorMessage=str(ex))
+
+# This gets the client after assuming the Config service role
+# either in the same AWS account or cross-account.
+def get_client(service, event):
+    """Return the service boto client. It should be used instead of directly calling the client.
+
+    Keyword arguments:
+    service -- the service name used for calling the boto.client()
+    event -- the event variable given in the lambda handler
+    """
+    if not ASSUME_ROLE_MODE:

```

Comment:
The GLOBAL variable is missing. It is in the template at the top.

Reply:
Add ASSUME_ROLE_MODE at line52.